webapp/channels/src/components/threading/global_threads/thread_menu/thread_menu.tsx

```diff
@@ -84,6 +88,22 @@ function ThreadMenu({
         unreadTimestamp,
     ]);
 
+    // TODO: This should be in a reusable component but since this menu hasn't been
```

Comment:
I know this is a TODO, but you could make this a reusable hook like `const [canPopout, popoutThread] = usePopoutThread(teamId, threadId);` instead of making a reusable menu item component. That may not even need the `teamId` parameter if we can just use the current team ID everywhere. That would let you still encapsulate this logic without tying it to a specific menu or button

Reply:
I would if all of the components using it were functional components. Sadly they are not :(